components/ILIAS/Authentication/classes/Logout/class.ilAuthLogoutBehaviourGUI.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+use ILIAS\Data\Link;
+use ILIAS\UI\Factory as UIFactory;
+use ILIAS\UI\Renderer as UIRenderer;
+use ILIAS\HTTP\Services as HttpService;
+use ILIAS\Refinery\Factory as Refinery;
+use ILIAS\UI\Component\Input\Container\Form\Standard as StandardForm;
+use ILIAS\Data\URI;
+
+/**
+ * @ingroup           ServicesAuthentication
+ * @ilCtrl_isCalledBy ilAuthLogoutBehaviourGUI: ilObjAuthSettingsGUI
+ * @ilCtrl_Calls      ilAuthLogoutBehaviourGUI: ilLoginPageGUI
+ */
+class ilAuthLogoutBehaviourGUI
+{
+    private int $ref_id;
+    private ilCtrl $ctrl;
+    private ilLanguage $lng;
+    private ilAccess $access;
+    private HttpService $http;
+    private Refinery $refinery;
+    private ilSetting $settings;
+    private UIFactory $ui_factory;
+    private UIRenderer $ui_renderer;
+    private ilGlobalTemplateInterface $tpl;
+
+    public function __construct(int $ref_id)
+    {
+        global $DIC;
+        $this->ref_id = $ref_id;
+        $this->ctrl = $DIC->ctrl();
+        $this->http = $DIC->http();
+        $this->lng = $DIC->language();
+        $this->access = $DIC->access();
+        $this->refinery = $DIC->refinery();
+        $this->tpl = $DIC->ui()->mainTemplate();
+        $this->ui_factory = $DIC->ui()->factory();
+        $this->ui_renderer = $DIC->ui()->renderer();
+        $this->lng->loadLanguageModule('auth');
+        $this->settings = new ilSetting('auth');
+    }
+
+    public function executeCommand(): void
+    {
+        $cmd = $this->ctrl->getCmd() . 'Cmd';
+        if (method_exists($this, $cmd)) {
+            $this->$cmd();
+        }
+        $this->showForm();
+    }
+
+    public function getForm(): StandardForm
+    {
+        $logout_group = $this->ui_factory->input()->field()->group(
+            [],
+            $this->lng->txt('destination_logout_screen')
+        );
+
+        $login_group = $this->ui_factory->input()->field()->group(
+            [],
+            $this->lng->txt('destination_login_screen'),
+            $this->lng->txt('destination_login_screen_info')
+        );
+
+        $ref_id = $this->ui_factory->input()->field()->numeric(
```

Comment:
We should add a "Custom Refinery Constraint" which validates the `$ref_id`. It MUST be a valid repository tree node which is NOT trashed, and it must be accessible by the anonymous user (which you already checked in `saveFormCmd`).

When populating the form with the data from the persistence layer, please add  a hint/info in case the `$ref_id` is NOT valid anymore, because the configured `$ref_id` got trashed/or completely deleted in the meantime.